### PR TITLE
feat(npm): Create a npm package for the data as well

### PR DESCRIPTION
**Pull request by steven-noorbergen**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  extends: [
+    "airbnb-typescript",
+    "airbnb/hooks",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:import/recommended",
+    "plugin:react/jsx-runtime",
+    "plugin:react/recommended",
+    "plugin:storybook/recommended",
+    "prettier",
```
Comment (TrueBrain) [inline .eslintrc.js:17]: This feels like total overkill for 2 lines of Typescript. Do we really need it?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  extends: [
+    "airbnb-typescript",
+    "airbnb/hooks",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:import/recommended",
+    "plugin:react/jsx-runtime",
+    "plugin:react/recommended",
+    "plugin:storybook/recommended",
```
Comment (TrueBrain) [inline .eslintrc.js:16]: Too much copy/paste .. pretty sure we don't have any storybook here (nor react, etc etc)

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  rules: {
+    "newline-per-chained-call": "off",
+    "react/jsx-pascal-case": "off",
+    "react/require-default-props": "off",
+  },
+  parserOptions: {
+    project: "./tsconfig.json",
+  },
+  ignorePatterns: ["src/EveDataProvider/esf_pb2.js", "src/EveDataProvider/protobuf.js"],
```
Comment (TrueBrain) [inline .eslintrc.js:33]: Ahum. Please check the rest of this file for too much copy/paste ;)

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -69,3 +86,15 @@ jobs:
         RCLONE_CONFIG_EVESHIPFIT_ACCESS_KEY_ID: ${{ secrets.R2_ACCESS_KEY_ID }}
         RCLONE_CONFIG_EVESHIPFIT_ENDPOINT: ${{ secrets.R2_ENDPOINT }}
         RCLONE_CONFIG_EVESHIPFIT_SECRET_ACCESS_KEY: ${{ secrets.R2_SECRET_ACCESS_KEY }}
+
+    # Build the package after pushing to R2 so that we don't push the javascript files to R2.
+    - name: Build package
+      run: |
+        npm run build
```
Comment (TrueBrain) [inline .github/workflows/release.yml:93]: I get the reasoning, but it also means building can fail, while the package is already deployed on R2. That is not ideal. So I rather have a JSON file too much ending up on R2, than having these kind of build failures. So just move this above the copying to R2 :)

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,11 @@
+For all data related to EVE Online please refer to the license in LICENSE.EVE. The license below applies to all EVEShipFit-specific code and data.
```
Comment (TrueBrain) [inline LICENSE:1]:
```suggestion
For all data related to EVE Online, please refer to the license in LICENSE.EVE. The license below applies to all EVEShipFit-specific code and data.
```

File: LICENSE.EVE [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+ELECTRONIC CONTRACTING
+
+By accepting this Agreement (including, without limitation, by accessing the Licensed Materials), you agree to transact business with CCP electronically. This applies to all transactions conducted through or involving the Licensed Materials. You may refuse to transact business with us at any time in the future by notifying us in writing and sending such notice to: CCP hf., Grandagardur 8, 101 Reykjavik, Iceland, Attention: General Counsel. You will need to maintain equipment, software and internet access necessary to access and use the features made available via the CCP Tools, as well as to request and access any copies of these terms and conditions, or CCP's Terms of Use or Privacy Policy. You may request a copy of each of these documents by contacting CCP at info@ccpgames.com provided you pay for the costs of copying and delivering the documents to you.
```
Comment (TrueBrain) [inline LICENSE.EVE:127]: Missing newline

File: list_shiptypes.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import os
+import sys
+import yaml
+import json
```
Comment (TrueBrain) [inline list_shiptypes.py:4]: abc. Please re-order :)

File: package.json [hunk 8]
```diff
@@ -1,5 +1,51 @@
...
+    "eslint-plugin-react": "^7.33.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-storybook": "^0.8.0",
+    "protobufjs-cli": "^1.1.2",
+    "rollup": "^4.4.0",
+    "rollup-plugin-dts": "^6.1.0",
+    "rollup-plugin-esbuild": "^6.1.0",
+    "rollup-plugin-node-externals": "^7.0.1",
+    "rollup-plugin-postcss": "^4.0.2",
+    "typescript": "^5.2.2"
```
Comment (TrueBrain) [inline package.json:49]: You also need to prune this list a bit ;) postcss .. react .. pretty sure this list should be nearly empty ;)

File: rollup.config.mjs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      },
+      {
+        file: "dist/esm/index.js",
+        format: "esm",
+        sourcemap: true,
+      },
+    ],
+    plugins: [
+      replace({
+        "process.env.CURRENT_VERSION": JSON.stringify(packageJson.version),
```
Comment (TrueBrain) [inline rollup.config.mjs:31]:
```suggestion
        "process.env.DATA_VERSION": JSON.stringify(packageJson.version),
```

"current" has no real meaning. I mean, it also isn't `CURRENT_BUILD_TIME` :P

File: rollup.config.mjs [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        "process.env.BUILD_TIME": JSON.stringify(new Date().toISOString()),
+      }),
+      nodeExternals(),
+      nodeResolve(),
+      commonjs(),
+      esbuild({ tsconfig: "./tsconfig.json" }),
+      postcss({ modules: true }),
```
Comment (TrueBrain) [inline rollup.config.mjs:38]: lol. Please minimalize your work :)

File: tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    "jsx": "react",
+    "module": "ESNext",
+    "moduleResolution": "node",
+    "outDir": "dist",
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "strict": true,
+    "target": "es2016"
+  },
+  "include": ["src/**/*.ts", "src/**/*.tsx", "src/**/*.d.ts", ".storybook/*.ts"],
```
Comment (TrueBrain) [inline tsconfig.json:18]: And here too, please remove stuff like storybook.

File: list_shiptypes.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        ships.append(
+            {
+                "id": id,
+                "name": name,
+                "group": groupName,
+            }
+        )
+
+with open("dist/ships.json", "w") as fp:
+    json.dump(ships, fp)
```
Comment (TrueBrain) [inline list_shiptypes.py:41]: This PR does much more than it suggests. Please split it up into two PRs, would also have made the review 1000 times easier. It is now a mix of adding new stuff content-wise, and changing infrastructure so an npm can be published. This makes it hard to distinguish what is what. Please don't :)

**Comment by steven-noorbergen** [comment]
Closing to split this up in different (cleaned up) PRs

**Pull request closed without merging**